hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordPayloadDelegate.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.model;
+
+import java.io.Serializable;
+import java.util.Properties;
+
+public interface HoodieRecordPayloadDelegate<T> extends Serializable {
```

Comment:
Can you please add a java-doc to explain purpose of this class?

Reply:
The idea of adding `HoodieRecordPayloadDelegate` is the same as the recording combining/merging API, i.e., decoupling the merging logic from record payload and adding an independent interface for merging, taking two type-generic instances and returning the result instance.

Reply:
since this class is not used in this PR and it'll probably be replaced with new API like `Mergeable`, I'll remove it from this patch.